### PR TITLE
OPCT-178: review periodics to add 4.10 and run in opct release

### PR DESCRIPTION
**Pull request by mtulio**
https://issues.redhat.com/browse/OPCT-178

The goals of this PR are:

- decrease the periodic to run all variations once a week only
- support to run in the binary of the`latest` release, not in the `main` OPCT
- prepare the periodic to run-upgrade mode, splitting the `install` and `upload` from ` run` (run-upgrade will be added separately)
- support regular executions on OCP 4.10
- updating owners

Dependencies:

- [x] https://github.com/openshift/release/pull/37955

Artifacts produced by rehearsal:

Screenshot from 2023-04-10 16-49-15

### CODE REVIEW

File: ci-operator/step-registry/provider-certification-tool/run/provider-certification-tool-run-commands.sh [hunk 1]
```diff
@@ -4,66 +4,9 @@ set -o nounset
...
-DATE=$(date +%Y%m%d)
-
-echo "s3://openshift-provider-certification/baseline-results/${VERSION}-${DATE}.tar.gz"
-echo '{"platform-type":"'"${CLUSTER_TYPE}"'"}'
-
-aws s3 cp "${ARTIFACT_DIR}"/certification-results/*.tar.gz "s3://openshift-provider-certification/baseline-results/${VERSION}-${DATE}.tar.gz" \
---metadata '{"platform-type":"'"${CLUSTER_TYPE}"'"}'
+# Run the tool in dedicated mode with watch flag set.
+echo "Running OPCT with regular mode"
+${OPCT_EXEC} run --watch
```
Comment (mtulio) [inline ci-operator/step-registry/provider-certification-tool/run/provider-certification-tool-run-commands.sh:12]: Allowing to log the stdout on CI to extract timing metrics

File: ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml [hunk 2]
```diff
@@ -1,7 +1,89 @@
 periodics:
 - agent: kubernetes
   cluster: vsphere
-  cron: 0 0 * * 2,4
+  cron: 0 0 * * 3
```
Comment (mtulio) [inline ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml:4]: nothing changed on 4.13, but the order of the jobs were sorted to the newest release first

**Comment by mtulio** [comment]
@rvanderp3 would you mind taking a look?
/assign @rvanderp3

**Comment by mtulio** [comment]
> The following users are mentioned in OWNERS file(s) but are untrusted for the following reasons. One way to make the user trusted is to add them as [members](https://github.com/orgs/openshift/people) of the openshift org. You can then trigger verification by writing `/verify-owners` in a comment.
> 
> * faermanj
>   
>   * User is not a member of the org. User is not a collaborator. Satisfy at least one of these conditions to make the user trusted.
>   * ci-operator/step-registry/provider-certification-tool/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/conf/dedicated/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/destroy/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/install/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/results/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/run/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/vsphere/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/vsphere/ipi/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/vsphere/upi-platform-none/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/vsphere/upi-platform-none/pre/OWNERS
>   * ci-operator/step-registry/provider-certification-tool/wait-ready/OWNERS

@faermanj would you mind requesting to add your username to the OCP org?

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-provider-certification-tool-vsphere-upi-platform-external

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
Sorry and thanks robert =]
/uncc bostrt

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.12-provider-certification-tool-vsphere-upi-platform-none

File: ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml [hunk 3]
```diff
@@ -178,13 +259,6 @@ periodics:
     ci.openshift.io/generator: prowgen
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none
-  reporter_config:
```
Comment (mtulio) [inline ci-operator/jobs/redhat-openshift-ecosystem/provider-certification-tool/redhat-openshift-ecosystem-provider-certification-tool-main-periodics.yaml:181]: Need to be fixed. reverted

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-provider-certification-tool-vsphere-upi-platform-external

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-provider-certification-tool-vsphere-upi-platform-external

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-provider-certification-tool-vsphere-upi-platform-external

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.12-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
Still fighting for OVA upload.

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
> /pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

```
Error: error cloning virtual machine: Post "https://ibmvcenter.vmc-ci.devcluster.openshift.com/sdk": EOF
```

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
refact install step to extract from image stream (awaiting to be merged on https://github.com/openshift/release/pull/37955)

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
Mirror PR merged, switching to use IS to extract OPCT image:

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
Now it's working correctly until the run step, suddenly I destroyed the old environment before finish, it was close to finishing the OCP conformance:

```
Thu, 06 Apr 2023 21:17:40 UTC> Global Status: running
JOB_NAME                           | STATUS     | RESULTS    | PROGRESS                  | MESSAGE                                           
05-openshift-cluster-upgrade       | complete   |            | 0/0 (0 failures)          | waiting for post-processor...                     
10-openshift-kube-conformance      | complete   |            | 352/352 (0 failures)      | waiting for post-processor...                     
20-openshift-conformance-validated | running    |            | 3278/3494 (15 failures)   | status=running=T/C/P/F/S=3494/3278/1222/15/2041   
99-openshift-artifacts-collector   | running    |            | 0/0 (0 failures)          | status=waiting-for=20-openshift-conformance-validated=(0/-216/0)=[1/1080]

```

Clean up the debug steps, moving the declaration and installation to the install step. Also standardizing the artifact object path considering the OPCT existing and future support:

~~~
# Examples:
# OPCT_VERSION/OCP_VERSION-DATE_TS-controlPlaneTopology-provider-platform.tar.gz
# v0.3.0/4.13.0-20230406-HighlyAvailable-vsphere-None.tar.gz
# v0.3.0/4.13.0-20230406-HighlyAvailable-aws-None.tar.gz
# v0.3.0/4.13.0-20230406-HighlyAvailable-aws-AWS.tar.gz
# v0.3.0/4.13.0-20230406-HighlyAvailable-oci-External.tar.gz
# v0.3.0/4.13.0-20230406-HighlyAvailable-oci-Baremetal.tar.gz
# v0.3.0/4.13.0-20230406-SingleReplica-aws-None.tar.gz
~~~

Re-running:

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.13-provider-certification-tool-vsphere-upi-platform-external

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.12-provider-certification-tool-vsphere-upi-platform-none

/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-4.12-provider-certification-tool-vsphere-upi-platform-none

**Comment by mtulio** [comment]
Final review:
/pj-rehearse periodic-ci-redhat-openshift-ecosystem-provider-certification-tool-main-provider-certification-tool-vsphere-upi-platform-none

**Comment by rvanderp3** [comment]
nice work @mtulio 
/lgtm

**Pull request merged**